subgraph/queries/harvests.py

```diff
@@ -7,7 +10,31 @@
 console = Console()
 
 
-def fetch_tree_distributions(start_timestamp, end_timestamp, chain):
+def _populate_end_of_previous_harvest(tree_distributions: List[Dict]):
+    """
+    This function groups distributions by strategy and adds
```

Comment:
This may lead to a problem when the strategy of the vault changes, it might be a better idea to group by vault instead. In the case of a new strategy deploy on the same vault the first harvest on the new strategy would be treated differently to how it normally is.

Reply:
Per Alex: `The new rule is to swap most of the times because upgrades are worse, and to upgrade when you can't swap` so we will need to be able to handle multiple strategy addresses for one vault.

So I think Swole is right here, we should be aggregating by vault, then iterating in increasing block number of harvests regardless of the strategy, as they would all be harvesting to the holders of the vault and should be considered in series.

Reply:
Done!